### ISSUE TITLE
Psuedoelements/classes not allowed in media queries

### ISSUE THREAD

**Issue by matthughes** [japgolly/scalacss#37]
```
  val foo = style("foo")(
    media.maxWidth(100 px)(
      &.hover {
        backgroundColor(yellow)
      }
    )
  )
```

Produces the following compiler error:

```
overloaded method value apply with alternatives:
[error]   (h: scalacss.AV,t: scalacss.AV*)scalacss.DslBase.ToStyle <and>
[error]   (avs: scalacss.AVs)scalacss.DslBase.ToStyle
[error]  cannot be applied to (scalacss.DslBase.ToStyle)
[error]     media.maxWidth(100 px)(
[error]                   ^
```

**Comment by matthughes** [comment]
Similarly, you can't solve this by inverting the terms:

```
  val foo = style("foo")(
    &.hover {
      media.maxWidth(100 px) {
        backgroundColor(yellow)
      }
    }
  )
```

gives the error:

```
overloaded method value apply with alternatives:
[error]   (h: scalacss.AV,t: scalacss.AV*)scalacss.DslBase.ToStyle <and>
[error]   (avs: scalacss.AVs)scalacss.DslBase.ToStyle
[error]  cannot be applied to (scalacss.DslBase.ToStyle)
[error]     &.hover {
```

**Comment by matthughes** [comment]
Ok, I read the docs about `Cond` but there are still a few wrinkles:

```
  val foo = style("foo") {
    (&.hover & media.maxWidth(100 px)) {
      backgroundColor(yellow)
    }
  }
```

This works, but note that you do have to put parens around the conditions or you will get a not helpful error message.

But reversing the conditions does not compile:

```
  val bar = style("bar") {
    (media.maxWidth(100 px) & &.hover) {
      backgroundColor(yellow)
    }
  }
```

produces:

```
[error]  found   : scalacss.Cond
[error]  required: scalacss.Media.Query
[error]     (media.maxWidth(100 px) & &.hover) {
[error]                                 ^
```


I think the latter should work no?  Also, I can put together a PR to show 1) more examples for the Cond page plus 2) have the examples use parens.

**Comment by matthughes** [comment]
Also, it would be nice to use &.hover and other psuedo elements as children of the query so as not to repeat oneself:

```
val bar = style("bar") {
  (media.maxWidth(100 px) & &.hover) {
    backgroundColor(yellow)
  },
  (media.maxWidth(100 px) & &.visited) {
    backgroundColor(blue)
  },
  (media.maxWidth(100 px) & &.active) {
    backgroundColor(red)
  }
}
```

vs:

```
val bar = style("bar") {
  (media.maxWidth(100 px)) {
    &.hover {
      backgroundColor(yellow)
    },
    &.visited {
      backgroundColor(blue)
    },
    &.active {
      backgroundColor(red)
    }
  }
}
```

I personally find the latter easier to read.

**Comment by japgolly** [comment]
Yes, this doesn't compile: `(media.maxWidth(100 px) & &.hover)`
But this does: `media.maxWidth(100 px).hover`

(Excluding your final comment for a second,) looking at it with fresh eyes, the DSL is a bit picky - you can the tell I was focused on functionality with the DSL slapped on last. I'm going to save a DSL overhaul for an eventual 1.0, for now the functionality *is* there. I think the doc isn't detailed enough. I'll add more combinations and attempt to clarify how to use it in these cases.

Your last point on repetition, seeing as it's all plain Scala, what about if you changed it to this.
```scala
  val mini = media.maxWidth(100 px)

  val bar = style("bar")(
    mini.hover(
      backgroundColor(yellow)),
    mini.visited(
      backgroundColor(blue)),
    mini.active(
      backgroundColor(red))
  )
```

What do you think?

**Comment by matthughes** [comment]
That works for me.  Actually better than any of the examples above as I can never keep the semantics of media queries in my head.

**Comment by japgolly** [comment]
Sweet, I'll update the doc to better demonstrate these combos

**Comment by japgolly** [comment]
Closed via bf86c6a
